vamb/__main__.py

```diff
@@ -1248,34 +1275,44 @@ def write_clusters_and_bins(
     clusters: dict[str, set[str]],
     sequence_names: Sequence[str],
     sequence_lens: Sequence[int],
+    to_file: bool = True,
+    unsplit_clusters_file: Optional[TextIO] = None,
+    split_clusters_file: Optional[TextIO] = None,
 ):
+    
     # Write unsplit clusters to file
     begintime = time.time()
-    unsplit_path = Path(base_clusters_name + "_unsplit.tsv")
-    with open(unsplit_path, "w") as file:
+    if unsplit_clusters_file:
         (n_unsplit_clusters, n_contigs) = vamb.vambtools.write_clusters(
-            file, clusters.items()
+            unsplit_clusters_file, clusters.items(), to_file
         )
-
+    else:
+        with open(unsplit_path, "w") as file:
+            (n_unsplit_clusters, n_contigs) = vamb.vambtools.write_clusters(
+                file, clusters.items(), to_file
+            )
+    
     # Open unsplit clusters and split them
     if binsplitter.splitter is not None:
-        split_path = Path(base_clusters_name + "_split.tsv")
         clusters = dict(binsplitter.binsplit(clusters.items()))
         # Add prefix before writing the clusters to file
         clusters = add_bin_prefix(clusters, bin_prefix)
-        with open(split_path, "w") as file:
+        if split_clusters_file:
```

Comment:
there are identical pieces of code handling `if split_clusters_file` and `if unsplit_clusters_file`. I would refactor it using a helper function that takes care of checking if the file exists:

```
from contextlib import nullcontext

def write_clusters_to_file(file_handle, file_path, clusters, to_file):
    handle = nullcontext(file_handle) if file_handle else open(file_path, 'w')
    with handle as file:
        return vamb.vambtools.write_clusters(file, clusters, to_file)
```

For info on `nullcontext`, check the docs https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext

Use this function to get rid of both `if split_clusters_file` and `if unsplit_clusters_file`